doc/coq.rst

```diff
@@ -494,6 +500,23 @@ All three of the theories we defined before were *private theories*. In order to
 depend on them, we needed to make them *public theories*. See the section on
 :ref:`public-private-theory`.
 
+Composing With Installed Theories
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can also compose with theories that are installed. If we wanted to have a
+theory that depends on the Coq theory ``mathcomp.ssreflect`` we can add the
+following to our stanza:
+
+.. code:: dune
+
+  (coq.theory
+   (name my_mathcomp_theory)
+   (theories mathcomp.ssreflect))
```

Comment:
What happens on overlapping paths? Can I use both `theories mathcomp` and `theories mathcomp.ssreflect` in the same project?

Reply:
Yup, you can even today, as long as you declared these theories of course.

Dune will just create the corresponding -Q flags and pass them to coqc.

For installed libraries, if the library is in user-contrib then dune has to infer the name, so it uses an approximation, it is on the user to keep sanity on the naming so the "install substitution lemma" does work, that's the price to pay to have 2-way compatibility as we aim (for now)

Reply:
I've added a note about `theories mathcomp` while in principle it will work, if mathcomp starts using a coq.theory stanza and you have a local copy it obviously won't find the correct name. Therefore I've mentioned that it is advisable to use `mathcomp.ssreflect`.